File: ANTERIOR/Projeto base - apCaminhosMarte/apCaminhosMarte/Cidade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Cidade : IComparable<Cidade>
{
    private int idCidade, coordenadaX, coordenadaY;
    private string nomeCidade;

    const int tamanhoId = 3;
    const int inicioNomeCidade = 3;
    const int tamanhoNomeCidade = 15;
    const int inicioX = inicioNomeCidade + tamanhoNomeCidade;
    const int tamanhoCoord = 5;
    const int inicioY = inicioX + tamanhoCoord;

    public Cidade(int id, int x, int y, string nome)
    {
        this.IdCidade = id;
        this.CoordenadaX = x;
        this.CoordenadaY = y;
        this.NomeCidade = nome;
    }

    public Cidade (string linha)
    {
        IdCidade    = int.Parse(linha.Substring(0, tamanhoId));
        NomeCidade = linha.Substring(inicioNomeCidade, tamanhoNomeCidade);
        CoordenadaX = int.Parse(linha.Substring(inicioX, tamanhoCoord));   
        CoordenadaY = int.Parse(linha.Substring(inicioY));
    }

    public int IdCidade { get => idCidade; set => idCidade = value; }
    public int CoordenadaX { get => coordenadaX; set => coordenadaX = value; }
    public int CoordenadaY { get => coordenadaY; set => coordenadaY = value; }
    public string NomeCidade { get => nomeCidade; set => nomeCidade = value; }

    public int CompareTo(Cidade other)
    {
        throw new NotImplementedException();
    }
}

File: 18169_18196_ProjetoED/app/src/main/java/br/unicamp/cotuca/pathbetweencities/cidade/Cidade.java
package br.unicamp.cotuca.pathbetweencities.cidade;

import androidx.annotation.NonNull;

import java.io.Serializable;

import br.unicamp.cotuca.pathbetweencities.interfaces.Copiavel;

public class Cidade implements Comparable<Cidade>, Copiavel<Cidade>, Serializable {
    private int idCidade;
    private float coordenadaX, coordenadaY;
    private String nomeCidade;

    protected final int tamanhoId = 2;
    protected final int inicioNomeCidade = 2;
    protected final int tamanhoNomeCidade = 16;
    protected final int inicioX = inicioNomeCidade + tamanhoNomeCidade;
    protected final int tamanhoCoord = 5;
    protected final int inicioY = inicioX + tamanhoCoord;

    public Cidade(int id, float x, float y, String nome)
    {
        this.setIdCidade(id);
        this.setCoordenadaX(x);
        this.setCoordenadaY(y);
        this.setNomeCidade(nome);
    }

    public Cidade (String linha)
    {
        setIdCidade(Integer.parseInt(linha.substring(0, tamanhoId).trim()));
        setNomeCidade(linha.substring(inicioNomeCidade, inicioNomeCidade+tamanhoNomeCidade).trim());
        setCoordenadaX(Float.parseFloat(linha.substring(inicioX, inicioX+tamanhoCoord).trim()));
        setCoordenadaY(Float.parseFloat(linha.substring(inicioY).trim()));
    }

    public int getIdCidade() {
        return idCidade;
    }

    public void setIdCidade(int idCidade) {
        this.idCidade = idCidade;
    }

    public float getCoordenadaX() {
        return coordenadaX;
    }

    public void setCoordenadaX(float coordenadaX) {
        this.coordenadaX = coordenadaX;
    }

    public float getCoordenadaY() {
        return coordenadaY;
    }

    public void setCoordenadaY(float coordenadaY) {
        this.coordenadaY = coordenadaY;
    }

    public String getNomeCidade() {
        return nomeCidade;
    }

    public void setNomeCidade(String nomeCidade) {
        this.nomeCidade = nomeCidade;
    }

    @Override
    public int compareTo(Cidade cidade) {
        return 0;
    }

    @Override
    public Cidade copia() {
        Cidade c = new Cidade(this.getIdCidade(), this.getCoordenadaX(), this.getCoordenadaY(), this.getNomeCidade());
        return c;
    }

    @NonNull
    @Override
    public String toString() {
        String ret = String.format("%-2s",this.getIdCidade()) +
                String.format("%-16s",this.getNomeCidade()) + String.format("%5s", this.getCoordenadaX()) + String.format("%6s", this.getCoordenadaY());
        return ret;
    }
}
